specs-go/parser.go

```diff
@@ -0,0 +1,39 @@
+/*
+   Copyright © The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package specs
+
+import "strings"
+
+// ParseQualifier splits a device qualifier into vendor and class.
```

Comment:
It's out of scope for this PR, but we may want to be stricter on our language here. In the spec the entity we're parsing here is the `Kind`, but this public function refers to it as a qualifier. Since we're in the `specs` package here, do we want to rename this to `SplitKind(string)` instead?

Reply:
Generally it makes sense to me. I'd propose to rename it to `ParseKind` if you don't mind. And I agree, it's out of scope of this PR.

Reply:
OK. Let's do this as a follow-up then.